### ISSUE TITLE
ACME plugin certificate autorenewal: failed to run timer 

### ISSUE THREAD

**Issue by Igor-lkm** [Kong/kong#13370]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

3.6.1

### Current Behavior

I found a similar issue here https://github.com/Kong/kong/issues/12442 but is says that it should be resolved in version 3.6

Certificates are not renewed automatically with `renew_threshold_days` or with `curl http://localhost:8001/acme -XPATCH`. 

Error messages in log:
```
2024/07/12 11:06:36 [error] 1322#0: *660849 [lua] job.lua:284: execute(): [timer-ng] failed to run timer unix_timestamp=1720782396775.000000;counter=912004:meta=debug off: /usr/local/share/lua/5.1/kong/plugins/acme/client.lua:497: attempt to index local 'config' (a boolean value), context: ngx.timer
```

However if i open a domain with an invalid or no certificate - then a renewal process is fine. First time I have a certificate warning in a browser and then it is fine after a reload, so a certificate for domain renewed successfully in this case.

Thank you for the help!

### Expected Behavior

Certificates are renewed automatically with the threshold or with `curl` request

### Steps To Reproduce

1) I use official kong helm chart on k8s
```
> helm list
NAME            	NAMESPACE	REVISION	UPDATED                              	STATUS  	CHART              	APP VERSION
kong   	gateway  	4       	2024-04-18 14:21:26.085576 +0200 CEST	deployed	kong-2.38.0        	3.6
```

2) Plugin is activated, and all necessary endpoint are added for `/.well-known/acme-challenge` 

```
curl http://localhost:8001/plugins \
   -d name=acme \
   -d config.account_email=test@test.com \
   -d 'config.domains[]=somedomain1.test.com' \
   -d 'config.fail_backoff_minutes=2' \
   -d 'config.renew_threshold_days=30`
```

Then i updated plugin multiple times, so final would look like:

```
curl -X PATCH http://localhost:8001/plugins/5096c070-35ba-479a-ad44-857c59cc09ca \
   -d name=acme \
   -d config.account_email=test@test.com \
   -d config.tos_accepted=true \
   -d 'config.domains[]=somedomain1.test.com' \
   -d 'config.domains[]=somedomain2.test.com' \
   -d 'config.fail_backoff_minutes=2' \
   -d 'config.renew_threshold_days=45' \
   -d 'config.storage=kong'
```

3) Try to renew
```
curl http://localhost:8001/acme -XPATCH
```

4) I got an error:
```
2024/07/12 11:06:36 [error] 1322#0: *660849 [lua] job.lua:284: execute(): [timer-ng] failed to run timer unix_timestamp=1720782396775.000000;counter=912004:meta=debug off: /usr/local/share/lua/5.1/kong/plugins/acme/client.lua:497: attempt to index local 'config' (a boolean value), context: ngx.timer
```

### Anything else?

```
KONG Version	3.6.1
LUA Version	LuaJIT 2.1.0-20231117
acme plugin	version	3.6.1, priority	1705
```

**Comment by chronolaw** [comment]
@fffonion , would you mind to take a look at this? thanks.

**Comment by fffonion** [comment]
cc @Water-Melon

**Comment by Water-Melon** [comment]
@Igor-lkm I just checked, and the content of that fix PR (#12773 ) is not included in Kong 3.6.1 (https://github.com/Kong/kong/blob/3.6.1/kong/plugins/acme/api.lua#L128) but is included in 3.7.0 (https://github.com/Kong/kong/blob/3.7.0/kong/plugins/acme/api.lua#L128).

**Comment by chronolaw** [comment]
@Igor-lkm , could you upgrade to kong gateway 3.7.x and verify again? thanks.
